core/src/main/java/com/kin/ecosystem/core/data/blockchain/BlockchainSourceImpl.java

```diff
@@ -98,20 +98,31 @@ public static BlockchainSourceImpl getInstance() {
 		return instance;
 	}
 
-	private void createKinAccountIfNeeded() throws BlockchainException {
-		int accountIndex = local.getAccountIndex();
+	private void initAccount() {
 		if (kinClient.hasAccount()) {
-			account = kinClient.getAccount(accountIndex);
-		} else {
+			if (account == null) {
+				account = kinClient.getAccount(activeAccountIndex);
```

Comment:
get account could be  a bit heavy operation when the app just starts

Reply:
do we have to get the account in init phase?

Reply:
nop, i'll move it into createAccountIfNeeded